### PR TITLE
Disable flaky tests

### PR DESCRIPTION
**Pull request by and-rewsmith**
Ported #5468 to 1.1 with the following changes:
1. Nested test disables don't apply, so obviously not included
2. Looked a the analytics for dps and x509 manual provisioning tests. Turns out there was some narrow blip of failures and then these tests have been quite stable. So I don't think we should disable them. I think we should re-enable in master / 1.2.

### CODE REVIEW

**Comment by and-rewsmith** [comment]
Test running here:
https://dev.azure.com/msazure/One/_build/results?buildId=46594806&view=results

Will run a few times and if all are green then I will merge

File: builds/e2e/templates/e2e-run.yaml [hunk 1]
```diff
@@ -10,14 +18,13 @@ steps:
...
         $(Join-Path $env:ProgramFiles 'iotedge-moby') -join ';'
-
       if ('$(minimal)' -eq 'true')
       {
         dotnet test $testFile --no-build --logger 'trx' --filter 'Name~TempSensor'
       }
       else
       {
-        dotnet test $testFile --no-build --logger 'trx'
+        dotnet test $testFile --no-build --logger 'trx' --filter "$filter"
```
Comment (and-rewsmith) [inline builds/e2e/templates/e2e-run.yaml:27]: Will make this filter logic apply to all cases

**Comment by and-rewsmith** [comment]
Building new images to confirm this works with the alteration to filtering.
https://dev.azure.com/msazure/One/_build/results?buildId=46595764&view=results

After I run the tests on these bits and they pass I will merge.

**Review by damonbarry — APPROVED** [review]

**Review by damonbarry — APPROVED** [review]

File: edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.E2E.Test/BackupAndRestoreTest.cs [hunk 2]
```diff
@@ -104,7 +104,7 @@ async Task BackupAndRestoreCorruptBackupMetadataTest(ITransportSettings[] transp
             Console.WriteLine("Finished test BackupAndRestoreCorruptBackupMetadataTest");
         }
 
-        [Theory]
+        [Theory(Skip = "Flaky")]
```
Comment (and-rewsmith) [inline edge-hub/test/Microsoft.Azure.Devices.Edge.Hub.E2E.Test/BackupAndRestoreTest.cs:107]: Tests are disabled to match 1.2 branch:
https://github.com/Azure/iotedge/blob/release/1.2/edge-hub/core/test/Microsoft.Azure.Devices.Edge.Hub.E2E.Test/BackupAndRestoreTest.cs#L107

**Review by damonbarry — APPROVED** [review]

**Pull request merged**